Source/DataSources/KmlDataSource.js

```diff
@@ -255,6 +255,23 @@ define([
         return deferred.promise;
     }
 
+    function insertNamespaces(text) {
+        var namespaceMap = {
+            'xsi:' : ['xmlns:xsi=', '"http://www.w3.org/2001/XMLSchema-instance"']
+        };
+        var firstPart;
+        var lastPart;
+
+        for (var key in namespaceMap){
```

Comment:
XML doesn't allow for whitespace between `<` and the tag name, so you could check for `<xsi:`, so you don't inadvertantly match soemthing like `<exsi:name>`. In the loop you could check for a bad file like this
```
var prefix = '<' + key + ':';
var declaration = 'xmlns:' + key + '=';
if(text.indexOf(prefix) !== -1 && text.indexOf(declaration]) === -1) {
```